packages/react-native-jigsaw/src/components/Image.tsx

```diff
@@ -13,37 +14,86 @@ import {
   GROUPS,
 } from "../core/component-types";
 
+import AspectRatio from "./AspectRatio";
+
+type ImageStyleProp = {
+  width?: number;
+  height?: number;
+  aspectRatio?: number;
+};
+
+const generateDimensions = ({ aspectRatio, width, height }: ImageStyleProp) => {
+  if (aspectRatio && !width && !height) {
+    return {
+      aspectRatio,
+      width: "100%",
+    };
+  }
+
+  if (aspectRatio && width && height) {
+    return {
+      aspectRatio,
+      height,
+      width: aspectRatio * height,
+    };
+  }
```

Comment:
can you eliminate this full block of code since it's covered by the next case?

Reply:
wdym?

Reply:
does it do the same thing, for sure?

Reply:
as far as i can tell you could remove lines 33 through 39 and it would work the same since when `aspectRatio && width && height` is true, `aspectRatio && height`. but if keeping it helps keep things more explicit, works for me

Reply:
actually it's not equivalent is it